src/core/packages/saved-objects/migration-server-internal/src/create_cummulative_logger.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { Logger, LogLevelId, LogMessageSource, LogMeta, LogRecord } from '@kbn/logging';
+import type { MigrationLog } from './types';
+
+export function createCummulativeLogger(logger: Logger): Logger & { dump: () => void } {
+  const migrationLogs: MigrationLog[] = [];
+
+  return {
+    debug<Meta extends LogMeta = LogMeta>(message: LogMessageSource, meta?: Meta) {
+      logger.debug<Meta>(message, meta);
+    },
+    trace<Meta extends LogMeta = LogMeta>(message: LogMessageSource, meta?: Meta) {
+      logger.trace<Meta>(message, meta);
+    },
+    fatal<Meta extends LogMeta = LogMeta>(message: LogMessageSource, meta?: Meta) {
+      logger.fatal<Meta>(message, meta);
+    },
+    log(record: LogRecord) {
+      logger.log(record);
+    },
```

Comment:
No need to queue these ones since they are not used in the migration steps. `debug` is used by v2 in the preparative steps.